lib/tether/ops_linux.go

```diff
@@ -267,22 +267,33 @@ func renameLink(t Netlink, link netlink.Link, slot int32, endpoint *NetworkEndpo
 	return link, nil
 }
 
-func getDynamicIP(t Netlink, link netlink.Link, dc client.Client) (*dhcp.Packet, error) {
+func getDynamicIP(t Netlink, link netlink.Link, endpoint *NetworkEndpoint) (client.Client, error) {
 	var ack *dhcp.Packet
 	var err error
 
 	// use dhcp to acquire address
-	id, err := client.NewID(link.Attrs().Index, link.Attrs().HardwareAddr)
+	dc, err := client.NewClient(link.Attrs().Index, link.Attrs().HardwareAddr)
 	if err != nil {
 		return nil, err
 	}
 
-	ack, err = dc.Request(id)
+	params := []byte{
+		byte(dhcp4.OptionSubnetMask),
```

Comment:
these are untyped constants, casting them to a specific type doesn't make much sense.